### PR TITLE
Implement new toast UI

### PR DESCRIPTION
**Pull request by weeman1337**
[New UI here.](https://www.figma.com/file/ja1Hf2UguAaD7ne1sYforo/DINUM-Feedback---Feb-2023?node-id=4%3A7863&t=tkIt0sNrL6tEUirB-4)

closes https://github.com/vector-im/element-web/issues/24490

Some toasts:

| Before | After |
| --- | --- |
| toast_before_verification |  toast_after_verification |
| call_before | image |
| new_login_before | new_login_after |

PSF-1924

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## ✨ Features
 * Implement new toast UI ([\#10467](https://github.com/matrix-org/matrix-react-sdk/pull/10467)). Fixes vector-im/element-web#24490.

### CODE REVIEW

File: res/css/structures/_ToastContainer.pcss [hunk 1]
```diff
@@ -138,22 +132,17 @@ limitations under the License.
...
 
         .mx_Toast_description {
-            max-width: 272px;
-            overflow: hidden;
-            text-overflow: ellipsis;
-            margin: 4px 0 11px 0;
-            font-size: $font-12px;
+            color: $primary-content;
+            font-size: $font-15px;
+            font-weight: 600;
```
Comment (kerryarchibald) [inline res/css/structures/_ToastContainer.pcss:144]: Could use `$font-semi-bold`

File: res/css/structures/_ToastContainer.pcss [hunk 2]
```diff
@@ -162,6 +151,10 @@ limitations under the License.
 
         .mx_Toast_detail {
```
Comment (kerryarchibald) [inline res/css/structures/_ToastContainer.pcss:152]: I think this could use the typography component `Caption`

File: res/css/structures/_ToastContainer.pcss [hunk 3]
```diff
@@ -104,24 +103,19 @@ limitations under the License.
...
             align-items: center;
+            box-sizing: border-box;
             column-gap: 8px;
+            display: flex;
+            margin-bottom: $spacing-16;
             width: 100%;
-            box-sizing: border-box;
 
             h2 {
+                color: $primary-content;
```
Comment (kerryarchibald) [inline res/css/structures/_ToastContainer.pcss:115]: could this use `Heading` from typography to avoid repeating these styles?

**Review by kerryarchibald — APPROVED** [review]
LGTM

File: res/css/structures/_ToastContainer.pcss (hunk 3 above)
Comment (weeman1337) [reply res/css/structures/_ToastContainer.pcss:115]: I will keep the custom heading because it doesn't match the UI design.

**Comment by rufuskahler** [comment]
Looks good @weeman1337 – However, for the Voice Call instance, can the buttons also be right aligned? Button padding here also looks slightly more generous. CC @gaelledel for VoIP perspective.

**Comment by weeman1337** [comment]
> Looks good @weeman1337 – However, for the Voice Call instance, can the buttons also be right aligned? Button padding here also looks slightly more generous. CC @gaelledel for VoIP perspective.

I can have another look. The [target issue](https://github.com/vector-im/element-web/issues/24490) was only about the verification toast. Call does use a custom toast implementation.

**Comment by weeman1337** [comment]
@rufuskahler I've updated the call toast to at least match the basic toast styles.

image

(see description to compare it to the currently existing one).

**Review by rufuskahler — APPROVED** [review]

**Comment by MTRNord** [comment]
image

this seems to have regressed the spacing of the update notification :/ The spacing is all over the place in it

**Comment by weeman1337** [comment]
> image
> 
> this seems to have regressed the spacing of the update notification :/ The spacing is all over the place in it

All toasts should now have more spacing in general according to the new UI.

**Comment by MTRNord** [comment]
> > image
> > this seems to have regressed the spacing of the update notification :/ The spacing is all over the place in it
> 
> All toasts should now have more spacing in general according to the new UI.

Well, my issue with that is that the toasts look weird/inconsistent now.

To point out a few issues with this toast specifically:

- The alignment is inconsistent. The text is left aligned while the buttons are right aligned. It would be nicer to either left align it all or center align the buttons. Or if right alignment is preferred, there should be a visual driven imho to make it flow better. This works better with the other ones since they have more width, making it visually obvious why they are aligned like that.
- The Text is inconsistent compared to the other toasts, it has 2 titles which are closely the same weight. This works nicer in the other toasts since they also have body weight text instead of just titles. Making it flow much nicer.
- Compared to the other toasts it is missing an icon. It imho makes an important difference with this new design. It helps the visual flow. The update toast is missing this entirely.

So in short the new design works for the other toasts but the update one feels left out and therfor inconsistent.

**Comment by MTRNord** [comment]
Oh, and I forgot but it feels very large vertically now. The "new version available" header feels too far away, imho. Making it feel artificially stretched (possibly due to flex shenanigans?)

**Comment by justjanne** [comment]
It *feels* like the new designs have a wider min-width for the toasts than the actually implemented toasts do?

**Comment by t3chguy** [comment]
image

The toasts feel broken IMO

**Comment by rufuskahler** [comment]
@weeman1337 – As discussed, let's revert back to the previous version until we can find more time to improve this, looking at all instances holistically and referencing the design system source files. @americanrefugee – Is probably the best person to do this. I can brief him next week on his return from leave.

**Comment by americanrefugee** [comment]
Here are 4 examples of a new toast UI for both Element X and Element Classic. I created these using 1 overall template to make things as easy to implement as possible ;)

[Here is the Figma file](https://www.figma.com/file/mq77SCq8h9hxw4VFqdm6pr/Toast-UI-for-Element-Classic?type=design&node-id=0-1).

Classic - Allow analytics
Classic - Incoming call
Classic - New sign in
Classic - Verification request
X - Allow analytics
X - Incoming call
X - New sign in
X - Verfication request

@rufuskahler @nadonomy What do you think?
